src/sidebar/helpers/mentions.ts

```diff
@@ -0,0 +1,81 @@
+import type { Mention } from '../../types/api';
+
+/**
+ * Wrap all occurrences of @mentions in provided text into the corresponding
+ * special tag, as long as they are surrounded by "empty" space (space, tab, new
+ * line, or beginning/end of the whole text).
+ *
+ * For example: `@someuser` with the `hypothes.is` authority would become
+ *  `<a data-hyp-mention data-userid="acct:someuser@hypothes.is">@someuser</a>`
+ */
+export function wrapMentions(text: string, authority: string): string {
+  return text.replace(/(?:^|\s)@(\w+)(?=\s|$)/g, (match, username) => {
+    const tag = document.createElement('a');
+
+    tag.setAttribute('data-hyp-mention', '');
+    tag.setAttribute('data-userid', `acct:${username}@${authority}`);
+    tag.textContent = `@${username}`;
+
+    return ` ${tag.outerHTML}`;
+  });
+}
+
+/**
+ * Replace all mentions wrapped in the special `<a data-hyp-mention />` tag with
+ * their corresponding plain-text representation
```

Comment:
```suggestion
 * their plain-text representation.
```

Just removing a word that I think is unnecessary.